src/DIRAC/Resources/Computing/BatchSystems/Host.py

```diff
@@ -222,11 +217,12 @@ def __checkPid(self, pid, user):
 
     if pid == 0:
       return "Unknown"
-    cmd = 'ps  -f -p %s | grep %s | wc -l' % (pid, user)
-    sp = subprocess.Popen(shlex.split(cmd), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    sp = subprocess.Popen(["ps", "-f", "-p", str(pid), "--no-headers"],
+                          stdout=subprocess.PIPE,
+                          stderr=subprocess.PIPE)
     output, error = sp.communicate()
     status = sp.returncode
-    if status == 0 and output.strip() == "1":
+    if status == 0 and len(output.split('\n')) == 2 and user in output:
```

Comment:
I don't think the length check really adds anything?
```suggestion
    if status == 0 and user in output:
```

Reply:
Same comment as above